backends/tc/README.md

```diff
@@ -0,0 +1,457 @@
+# Parser back-end for tc extensions
+
+This back-end compiles only the parser program (currently) to a ternary content
+addressable memory (TCAM)-based abstract machine, which is the most common
+mechanism for realizing packet parsers in hardware today. The TCAM-based machine
+is based on the ideas from [_Design Principles for Packet Parsers_ by Gibb et
+al.](http://yuba.stanford.edu/~nickm/papers/ancs48-gibb.pdf) (especially Section
+5 of the paper).
+
+The goal of this back-end is to fill a gap that currently exists among P4
+back-ends today. Most parser hardware is TCAM based, however all the backends
+(except bpf) are architecture agnostic.
+
+The resulting TCAM program is output as a YAML file. Each TCAM entry is
+expressed using an [extended tc-flower grammar](#tc-extensions).
+
+We have also implemented a software simulator for parsers written for the
+TCAM-based abstract machine. The simulator will be open-sourced soon.
+
+## Additional dependencies
+
+On top of the p4c dependencies, this back-end requires:
+
+- Abseil, following Abseil's Live at Head policy, the back-end is built and
+  tested
+- yaml-cpp 0.7.0
+
+All of the dependencies are specified in the Bazel build file
+(`/bazel/p4c_deps.bzl`).
+
+## Building and testing the back-end
+
+The back-end uses Bazel for building, in the root of the `p4c` repository, you
```

Comment:
I am working on setting up CMake (@fruffy gave some good pointers on how to set it up), I set it up to build locally. I will update the PR soon, once I set up the tests to run on CMake, and verify that it works on a clean Ubuntu Xenial image rather than my local computer's configuration.